util/make_new_dif.py

```diff
@@ -133,7 +150,30 @@ def main():
     args = parser.parse_args()
 
     # Parse CMD line args.
-    ip = Ip(args.ip, args.peripheral)
+    ips = []
+
+    # Check for batch generation mode (used in CI check:
+    # ci/scripts/check-generated.sh")
+    if args.mode == "batch":
+        assert (
+            len(args.only) == 1 and args.only[0] == "autogen" and
+            "ERROR: can only batch generate completely auto-generated code.")
```

Comment:
Shouldn't this be an `if not (my condition): raise RuntimeError(...)`? I don't think that the argparse stuff guarantees this, and an `assert` should mean a programming error, not a user error.

Similarly for the assertion below.

Reply:
Fixed.

Reply:
If you're doing a re-push, you might want to catch the assertion on line 168 too.